app/controllers/CyaCategorisationController.scala

```diff
@@ -105,6 +107,7 @@ class CyaCategorisationController @Inject() (
             model.categoryAssessmentsWithExemptions,
             model.category
           )
+          dataCleansingService.deleteMongoData(request.userAnswers.id)
```

Comment:
Should this happen after the goodsRecordConnector call? Just wondering if we want the data cleared if that fails?

Reply:
We would always want to clear it if it's a validation failure and in the scenario where it succeeds. I can see a scenario where it would be useful to keep it being if we get a 500 if B&T is down but I think for now this is okay/not over engineering/see if it comes up in the feedback

Reply:
That's a good point, let's move this inside the map of the connector calls

Reply:
moved it inside connector calls